src/common/hooks/useScrollHandler.tsx

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react';
+
+export const useScrollHandler = (
+  handler: () => void,
+  cleanup?: () => void,
+): void => {
+  useEffect(() => {
+    window.addEventListener('scroll', handler);
```

Comment:
with a effect, it must always have a dependency. If you pass an empty array as your dependency it will call the function on mount. The return function is called before being destroyed.